streams/actions/src/main/java/com/hortonworks/streamline/streams/actions/container/TopologyActionsContainer.java

```diff
@@ -101,15 +104,30 @@ private TopologyActions initTopologyActions(Map<String, Object> conf, String cla
 
         Component uiServer = getComponent(streamingEngineService, COMPONENT_NAME_STORM_UI_SERVER)
                 .orElseThrow(() -> new RuntimeException(streamingEngine + " doesn't have " + COMPONENT_NAME_STORM_UI_SERVER + " as component"));
-        String uiHost = uiServer.getHosts().get(0);
-        Integer uiPort = uiServer.getPort();
+
+        Collection<ComponentProcess> uiServerProcesses = environmentService.listComponentProcessesInComponent(uiServer.getId());
+        if (uiServerProcesses.isEmpty()) {
+            throw new RuntimeException(streamingEngine + " doesn't have any process for " + COMPONENT_NAME_STORM_UI_SERVER + " as component");
+        }
+
+        ComponentProcess uiServerProcess = uiServerProcesses.iterator().next();
+        String uiHost = uiServerProcess.getHost();
```

Comment:
Lots of local var can be final. Let's adopt style guide and integrate checkstyle to the build phase if we really want to handle style issue.